ccl/sqlproxyccl: TestConnectionMigration failed

ccl/sqlproxyccl.TestConnectionMigration [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5511598&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5511598&tab=artifacts#/) on release-22.1 @ [5e58d0ff8a596cd3ea9482a9b5a160fca82168a8](https://github.com/cockroachdb/cockroach/commits/5e58d0ff8a596cd3ea9482a9b5a160fca82168a8):


```
=== RUN   TestConnectionMigration/transfer_timeout_in_response
    --- FAIL: TestConnectionMigration/transfer_timeout_in_response (5.17s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-experience @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestConnectionMigration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-16861

--- a/pkg/ccl/sqlproxyccl/proxy_handler_test.go
+++ b/pkg/ccl/sqlproxyccl/proxy_handler_test.go
@@ -1280,10 +1280,8 @@ func TestConnectionMigration(t *testing.T) {
 		conn, err := db.Conn(tCtx)
 		require.NoError(t, err)
 
-		// Spin up a goroutine to trigger the initial connection.
-		go func() {
-			_ = conn.PingContext(tCtx)
-		}()
+		// Trigger the initial connection.
+		require.NoError(t, conn.PingContext(tCtx))
 
 		var f *forwarder
 		require.Eventually(t, func() bool {
@@ -1333,15 +1331,9 @@ func TestConnectionMigration(t *testing.T) {
 		// one test.
 		<-goCh
 		time.Sleep(2 * time.Second)
-		// This should be an error because the transfer timed out.
+		// This should be an error because the transfer timed out. Connection
+		// should automatically be closed.
 		require.Error(t, f.TransferConnection())
-
-		// Connection should be closed because this is a non-recoverable error,
-		// i.e. timeout after sending the request, but before fully receiving
-		// its response.
-		err = conn.PingContext(tCtx)
-		require.Error(t, err)
-		require.Regexp(t, "(closed|bad connection)", err.Error())
 
 		select {
 		case <-time.After(10 * time.Second):

